src/raglite/_extract.py

```diff
@@ -33,11 +33,17 @@ class MyNameResponse(BaseModel):
     # Load the default config if not provided.
     config = config or RAGLiteConfig()
     # Update the system prompt with the JSON schema of the return type to help the LLM.
-    system_prompt = (
-        return_type.system_prompt.strip() + "\n",  # type: ignore[attr-defined]
-        "Format your response according to this JSON schema:\n",
-        return_type.model_json_schema(),
+    system_prompt = "\n".join(
+        (
+            return_type.system_prompt.strip(),  # type: ignore[attr-defined]
+            "Format your response according to this JSON schema:",
+            str(return_type.model_json_schema()),
+        )
     )
+    response_format: dict[str, Any] = {"type": "json_object"}
```

Comment:
Is it not possible to specify the schema with the OpenAI API? If we do it this way I'm a bit concerned the output may not satisfy the JSON schema.

Reply:
It looks like it should be possible according to the LiteLLM docs: https://docs.litellm.ai/docs/completion/json_mode#pass-in-json_schema

Reply:
And I think we probably also want to enable schema validation: https://docs.litellm.ai/docs/completion/json_mode#validate-json-schema, so that the for loop will retry if it doesn't validate.

Reply:
right, i had missed this, it was added in the august for openai it seems. We used the json_mode in the past and it seems reliable, but clearly enforcing it is better.

I will have a look at this, but im not sure the schema format is the same as you are using currently.